internal/observability/metrics.go

```diff
@@ -0,0 +1,57 @@
+package observability
+
+import (
+	"context"
+	"sync"
+)
+
+type contextKey string
+
+const keyMetricsRecorder contextKey = "metricsRecorder"
+
+type (
+	Metric interface {
+		IsMetric()
+	}
+
+	MetricsRecorder interface {
+		Metrics() []Metric
```

Comment:
I think we need it because the `slabMover` returns `[]HostInteraction` now. It doesn't really make sense to use an `io.Writer` here? Also we currently don't write the metrics to a log file we can always go in an make that work.